keras_nlp/samplers/sampler.py

```diff
@@ -358,22 +367,30 @@ def one_step(
                 return current_index, prompt, mask
             return [current_index, prompt, mask, cache]
 
+        def cond(current_index, prompt, mask, cache=None):
+            if end_token_id is None:
+                return True
+            end_token_seen = (prompt == end_token_id) & (
+                original_padding_mask == 0
+            )
+            sequence_done = tf.reduce_any(end_token_seen, axis=-1)
+            all_done = tf.reduce_all(sequence_done)
+            return tf.logical_not(all_done)
```

Comment:
I think `not all_done` will compile just fine with and without XLA, and is a bit more readable. Though worth checking.